### PR TITLE
Qute type-safe templates - fix validation of expressions in set sections

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

File: independent-projects/qute/core/src/main/java/io/quarkus/qute/SetSectionHelper.java [hunk 1]
```diff
@@ -68,7 +70,12 @@ public Scope initializeBlock(Scope previousScope, BlockInfo block) {
                 Scope newScope = new Scope(previousScope);
                 for (Entry<String, String> entry : block.getParameters().entrySet()) {
                     Expression expr = block.addExpression(entry.getKey(), entry.getValue());
-                    newScope.putBinding(entry.getKey(), expr.collectTypeInfo());
+                    //newScope.putBinding(entry.getKey(), expr.collectTypeInfo());
```
Comment (gsmet) [inline independent-projects/qute/core/src/main/java/io/quarkus/qute/SetSectionHelper.java:73]: Why the comment?

Comment (mkouba) [reply independent-projects/qute/core/src/main/java/io/quarkus/qute/SetSectionHelper.java:73]: Good catch. It should not be there at all...

File: extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java [hunk 2]
```diff
@@ -1195,20 +1194,31 @@ static boolean processHints(TemplateAnalysis templateAnalysis, String helperHint
...
+                if (valueExprMatch != null) {
+                    match.setValues(valueExprMatch.clazz, valueExprMatch.type);
+                }
+            }
         }
         return false;
     }
 
+    private static Expression findExpression(String helperHint, String hintPrefix, TemplateAnalysis templateAnalysis) {
+        Integer id = Integer.valueOf(helperHint.substring(hintPrefix.length(), helperHint.length() - 1));
```
Comment (gastaldi) [inline extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java:1218]: Change it to `Integer.parseInt` and remove the unnecessary Integer autoboxing in `io.quarkus.qute.deployment.TemplatesAnalysisBuildItem.TemplateAnalysis#findExpression`

**Review by gastaldi — APPROVED** [review]
Awesome!

**Pull request merged**